src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -761,21 +887,26 @@ private Task WriteSuffix()
                 responseHeaders.SetRawConnection("keep-alive", _bytesConnectionKeepAlive);
             }
 
-            if (ServerOptions.AddServerHeader && !responseHeaders.HasServer)
-            {
-                responseHeaders.SetRawServer(Constants.ServerName, _bytesServer);
-            }
+            EmitHttp11(ref end);
+            WriteStatus(ref end);
+            responseHeaders.CopyTo(ref end);
 
             if (!responseHeaders.HasDate)
             {
                 var dateHeaderValues = DateHeaderValueManager.GetDateHeaderValues();
                 responseHeaders.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
```

Comment:
I don't think that the date needs to be added to the response headers any more.

On the same note, I don't think we need to generate the string anymore, so GetDateHeaderValues can just return a byte array instead of DateHeaderValues. This also means we don't need to make Date a "raw" header anymore.